documentation/api/query/v4/index.markdown

```diff
@@ -0,0 +1,66 @@
+---
+title: "PuppetDB 3.2 » API » v4 » Root Endpoint"
+layout: default
+canonical: "/puppetdb/latest/api/query/v4/index.html"
+---
+
+[curl]: ../curl.html#using-curl-from-localhost-non-sslhttp
+[paging]: ./paging.html
+[query]: ./query.html
+[from]: ./operators.html#context-operators
+[entities]: ./entities.html
+
+*Note:* This endpoint is experimental. It may be altered or removed in a future release.
+
+The root query endpoint can be used to retrieve any known entities, from a
+single endpoint.
+
+## `/pdb/query/v4`
+
+This will return any known entity based on the required `query` field. Unlike
+other endpoints, the [entity][entities] must be supplied using a query with the [`from`][from]
+operator.
+
+### URL Parameters
+
+* `query`: Required. A JSON array containing the query in prefix notation
+(`["from", "<ENTITY>", ["<OPERATOR>", "<FIELD>", "<VALUE>"]]`). See the
+sections below for the supported operators and fields. Unlike other endpoints,
+a query with a `from` is required to choose the entity to query for. For
+general info about queries, see [the page on query structure.][query]
+
+If a query parameter is not provided, all results will be returned.
+
+### Response Format
+
+The response will be in `application/json`, and will contain a list of JSON
+object results based on the query provided.
```

Comment:
I'll link to the entities page a little better. I'm avoiding repeating myself here, since that list will become out of date. Sometimes, we need the ability to insert sub-documents, rather than link to them I think.